internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -1041,6 +1044,11 @@ func (r *oauthClientResource) ModifyPlan(ctx context.Context, req resource.Modif
 		}
 	}
 
+	if plan.AllowAuthenticationApiInit.ValueBool() && internaltypes.IsDefined(plan.RestrictScopes) && !plan.RestrictScopes.ValueBool() {
```

Comment:
If plan.RestrictScopes is non-Unknown you aren't allowed to change the value here, since the true value won't match the config.